kafka-rest/src/main/java/io/confluent/kafkarest/ProducerMetrics.java

```diff
@@ -0,0 +1,396 @@
+   * @param tags pairs of tag name and values
+   * @return the {@link ProduceMetricMBean} that can be used to create metrics; never null
+   */
+  public ProduceMetricMBean mbean(String groupName, Map<String, String> tags) {
+    BeanCoordinate beanCoordinate = new BeanCoordinate(groupName, tags);
+    beansByCoordinate.putIfAbsent(beanCoordinate, new ProduceMetricMBean(beanCoordinate));
+    return beansByCoordinate.get(beanCoordinate);
+  }
+
+  /** Stop and unregister the metrics from any reporters. */
+  public void stop() {
+    log.info("Unregistering produce metrics with JMX");
+    metrics.close();
+  }
+
+  /** Sets up a {@link ProduceMetricMBean} for each sensor */
+  private void setupMetricBeans() {
+    ProduceMetricMBean metricMBean =
+        mbean(ProducerMetricsRegistry.GROUP_NAME, Collections.emptyMap());
+
+    // Clear out any previous metrics and sensors that might have been present
+    if (metricMBean != null) {
+      metricMBean.close();
+    }
+
+    // request metrics
+    new SensorBuilder(metricMBean)
+        .withName(ProducerMetricsRegistry.REQUEST_RATE)
+        .withDoc(ProducerMetricsRegistry.REQUEST_RATE_DOC)
+        .withRate()
+        .build();
+
+    new SensorBuilder(metricMBean)
+        .withName(ProducerMetricsRegistry.REQUEST_SIZE_AVG)
+        .withDoc(ProducerMetricsRegistry.REQUEST_SIZE_AVG_DOC)
+        .withAvg()
+        .build();
+
+    new SensorBuilder(metricMBean)
+        .withName(ProducerMetricsRegistry.REQUEST_SIZE_TOTAL)
+        .withDoc(ProducerMetricsRegistry.REQUEST_SIZE_TOTAL_DOC)
+        .withCumulativeSum()
+        .build();
+
+    new SensorBuilder(metricMBean)
+        .withName(ProducerMetricsRegistry.REQUEST_TOTAL)
+        .withDoc(ProducerMetricsRegistry.REQUEST_TOTAL_DOC)
+        .withCumulativeCount()
+        .build();
+
+    // response metrics
+    new SensorBuilder(metricMBean)
+        .withName(ProducerMetricsRegistry.RESPONSE_SEND_RATE)
+        .withDoc(ProducerMetricsRegistry.RESPONSE_SEND_RATE_DOC)
+        .withRate()
+        .build();
+
+    new SensorBuilder(metricMBean)
+        .withName(ProducerMetricsRegistry.RESPONSE_SIZE_AVG)
+        .withDoc(ProducerMetricsRegistry.RESPONSE_SIZE_AVG_DOC)
+        .withAvg()
+        .build();
+
+    new SensorBuilder(metricMBean)
+        .withName(ProducerMetricsRegistry.RESPONSE_SIZE_TOTAL)
+        .withDoc(ProducerMetricsRegistry.RESPONSE_SIZE_TOTAL_DOC)
+        .withCumulativeSum()
+        .build();
+
+    new SensorBuilder(metricMBean)
+        .withName(ProducerMetricsRegistry.RESPONSE_TOTAL)
+        .withDoc(ProducerMetricsRegistry.RESPONSE_TOTAL_DOC)
+        .withCumulativeCount()
+        .build();
+
+    new SensorBuilder(metricMBean)
+        .withName(ProducerMetricsRegistry.RECORD_ERROR_RATE)
+        .withDoc(ProducerMetricsRegistry.RECORD_ERROR_RATE_DOC)
+        .withRate()
+        .build();
+
+    new SensorBuilder(metricMBean)
```

Comment:
I added a windowed counter via https://github.com/confluentinc/rest-utils/pull/284/files so that we can alert on absolute values for errors, rather than aggregated rates.  Might be worth doing the same here.

Reply:
@ehumber sorry for the delay, I've added the windowed metrics now.